### DISCUSSION TITLE
HTML diagram export and interactive SVG exports are missing from the new open source version

### DISCUSSION THREAD

**Discussion by simonbrowndotje** [structurizr/lite#34] [Announcements]
The HTML and interactive SVG exports are less useful now that the [diagram renderer can be used standalone](https://github.com/structurizr/ui/blob/main/examples/diagram-basic.html) ... leave a comment if you miss these features.

**Comment by gp-mlezotte** [comment]
I had just started using the HTML export, but my use case may not be worth adding the feature back in for. I'm having trouble getting my organization to stand up an on-prem instance that's accessible to everyone (especially external organizations) and am not permitted to use the cloud version. So, I've essentially been exporting via the HTML method to share with other organizations. This method is convenient since it still has navigation functionality available.

**Comment by simonbrowndotje** [comment]
There's a basic example of how to create your own offline diagram page, with double-click navigation, at https://github.com/structurizr/ui/blob/main/examples/diagram-navigate.html ... to be honest though, I'd just distribute the `workspace.json` file along with a script to start the Docker build of Structurizr Lite. This way you have everything running locally, with the benefit of the full interactive diagrams.

**Comment by gp-mlezotte** [comment]
Thanks for the tip @simonbrowndotje, we actually have been taking that approach internally (script + ```workspace.json``` distribution via git repo). I think if our workspaces were scoped a little better, we could probably take the same approach with external as well.

**Comment by rosborne21** [comment]
Hello @simonbrowndotje !  Big fan of C4 / Strucutrizr!  I have been spreading the word about diagrams as code all throughout my company.  Maybe there will be an opportunity for collaboration in the future.  

I have written some custom code to allow for interactive C4 diagrams in in [MkDocs](https://www.mkdocs.org/) using the Interactive SVG exports (includes breadcrumbs as you drill into the model).  I planned to eventually convert it to an MkDocs plugin and share it with the community.  However, in the mean time, I am dependent on the interactive SVG exports to get my near term work done.  Is there anyway you can enable that feature again?  Or tell me what's the latest release that supports interactive SVG exports.  

Thank you!  

V/R,
Rick

**Comment by morquan** [comment]
Greetings,

I definitely found the single file HTML export valuable and have been missing it. There are various scenarios where documentation needs to be distributed to people who don't have access to shared systems for one reason or another. I appreciate that there's a workaround but I think this will be harder to discover for future users.

Thanks for the great tool!

**Comment by mikhail-rush** [comment]
How can I enable tooltips in HTML diagrams?
I've tried `diagram.setTooltip(true)` but I'm getting:
```
structurizr-diagram.js:435 Uncaught TypeError: tooltip.hide is not a function
    at renderView (structurizr-diagram.js:435:21)
    at structurizr.ui.Diagram.changeView (structurizr-diagram.js:377:9)
    at showSelectedDiagram (diagram-navigate.html:2376:15)
    at diagram-navigate.html:2360:11
```
